### PR TITLE
Upgrade to Clang15 and GCC9 on self-hosted CI

### LINKED ISSUE #4242: Bump minimal CI compilers versions
**Is your feature request related to a problem? Please describe.**
QMCPACK has a two-year rule support for minimal compiler versions supported. Some GCC compilers toolchains in CI predate this two year limit.

**Describe the solution you'd like**
Bump minimum CI version on sulfur, nitrogen and migrate to Ubuntu 22 containers for GitHub Actions CI

**Describe alternatives you've considered**

**Additional context**
Related to PR #4236 

- [ ] use and test gcc-9 on sulfur and nitrogen, RHEL8 uses gcc-8 by default

### PR DESCRIPTION
**Pull request by williamfgc**
## Proposed changes
Bump from RHEL8 system GCC8 to GCC9 on sulfur and nitrogen CI
Trying Clang15 on sulfur built pointing at GCC9
Fixes #4242 
Related to PR #4236 to make GCC9 a minimal requirement

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- Testing changes (e.g. new unit/integration/performance tests)

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
sulfur, needs to pass CI

## Checklist

- Yes. This PR is up to date with current the current state of 'develop'
- No. Code added or changed in the PR has been clang-formatted
- No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by williamfgc** [comment]
Test this please

**Comment by williamfgc** [comment]
TO DO: need to update job names to GCC9 on yaml file after current CI passes with configure logs catching GCC9.

**Comment by ye-luo** [comment]
Test this please

File: tests/test_automation/github-actions/ci/run_step.sh [hunk 1]
```diff
@@ -375,7 +383,7 @@ case "$1" in
 
     if [[ "${GH_JOBNAME}" =~ (AFQMC-Offload) ]]
     then
-       export LD_LIBRARY_PATH=/opt/llvm/15.0.0-rc3/lib:/usr/lib64/openmpi/lib:${LD_LIBRARY_PATH}
+       export LD_LIBRARY_PATH=/opt/llvm/15.0.0/lib:/usr/lib64/openmpi/lib:${LD_LIBRARY_PATH}
```
Comment (prckent) [inline tests/test_automation/github-actions/ci/run_step.sh:386]: Does not need fixing here, but clearly this openmpi setting is in the wrong location; there should be an "if MPI" section.

**Comment by prckent** [comment]
Ye beat me to the error bar update. This is fine.

**Comment by ye-luo** [comment]
> TO DO: need to update job names to GCC9 on yaml file after current CI passes with configure logs catching GCC9.

How about just call tests GCC without version numbers? Then we don't need to update job yaml all the time.
We can figure out the version from the configure stage cmake printout anyway.

**Comment by prckent** [comment]
I prefer to have the version number in the job name since it is a regular reminder of the different compiler versions that we are testing and helpful for people who otherwise might be unsure where to look. It is also consistent with the nightlies. (In those we can figure out the version of the compiler and set the cdash submit name automatically.)

File: tests/test_automation/github-actions/ci/run_step.sh (hunk 1 above)
Comment (williamfgc) [reply tests/test_automation/github-actions/ci/run_step.sh:386]: If I recall correctly AFQMC always needs MPI that was the logic behind this if. We can change it.

**Comment by williamfgc** [comment]
After merging we can disable the old `GCC8-*` required statuses and move to the new `GCC9-*`. Thanks!

**Comment by ye-luo** [comment]
@prckent please force merge this PR. There is no more GCC8 configuration support and ornl-ci expect to fail.

**Comment by prckent** [comment]
I'll have to update the required checks when these new ones report on an open PR.

**Pull request merged**